src/index.ml

```diff
@@ -409,56 +512,86 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
     (go [@tailcall]) 0L 0 0
 
   let merge ~witness t =
-    Log.debug (fun l -> l "unforced merge %S\n" t.root);
-    let log = assert_and_get t.log in
-    let merge_path = merge_path t.root in
-    let generation = Int64.succ t.generation in
-    let log_array =
-      let compare_entry e e' = compare e.key_hash e'.key_hash in
-      let b = Array.make (Tbl.length log.mem) witness in
-      Tbl.fold
-        (fun key value i ->
-          b.(i) <- { key; value; key_hash = K.hash key };
-          i + 1)
-        log.mem 0
-      |> ignore;
-      Array.fast_sort compare_entry b;
-      b
-    in
-    let fan_size =
-      match t.index with
-      | None -> Tbl.length log.mem
-      | Some index ->
-          (Int64.to_int (IO.offset index.io) / entry_size) + Tbl.length log.mem
+    flush_instance t;
```

Comment:
Maybe you want to call `flush`(with the `rename_lock`) here and not `flush_instance`?